lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -622,6 +626,62 @@ dummySkeleton inputCount outputs = SelectionSkeleton
         TokenMap.empty
     }
 
+_evaluateMinimumFee
+    :: Cardano.ProtocolParameters
+    -> SealedTx
+    -> Coin
+_evaluateMinimumFee pp tx =
+    fromCardanoLovelace minFee
+  where
+    minFee = case getSealedTxBody tx of
+        InAnyCardanoEra ShelleyEra txbody ->
```

Comment:
I'm looking into this, but I don't think we need to expose the ledger txbodies here... The `Cardano.Tx body wits` pattern should be enough...

Reply:
Or is it intentional for some reason?

Reply:
The required `IsShelleyBasedEra era`, I think we should be able to conjure up somehow...

Reply:
unfortunately,in the case of tx certs I cannot extract from cardano-node due to how it sneaks BuildTx there. And it turn out that doing this through ledger tx is a way...

Reply:
feel free to try without ledger bypass if you have a while

Reply:
For the record, what I had missed until our quick call was:

- We need to predict how many witnesses there will be, as the transaction is not signed.
- We cannot use [estimateTransactionKeyWitnessCount](https://github.com/input-output-hk/cardano-node/blob/0ed04ab8d0a205691219f04623a3d880545a3f34/cardano-api/src/Cardano/Api/Fees.hs#L284-L285)